GameOfRunes/GameOfRunes/GameEngine/GameEngine.swift

```diff
@@ -251,19 +237,15 @@ class GameEngine {
         metadata.multiplier = 1.0
     }
     
-    func updateSelectedPowerUp() {
-        metadata.selectedPowerUp = gameScene?.selectedPowerUp
+    func updateSelectedPowerUp(powerUpType: PowerUpType?) {
+        metadata.selectedPowerUp = powerUpType
         
-        switch metadata.selectedPowerUp {
-        case .heroicCall:
+        switch metadata.selectedPowerUp?.activationType {
+        case .immediate:
             // although these power ups do not need position, position is set to center of screen
             // so that that messages will appear at the center if any
-            activatePowerUp(at: gameScene?.center
-                ?? .init(x: UIScreen.main.bounds.midX, y: UIScreen.main.bounds.midY))
-        case .divineShield:
-            activatePowerUp(at: gameScene?.playerEndPoint.position
-                ?? .init(x: UIScreen.main.bounds.midX, y: UIScreen.main.bounds.midY))
-        case .darkVortex:
+            activatePowerUp(at: gameScene?.center ?? .init(x: UIScreen.main.bounds.midX, y: UIScreen.main.bounds.midY))
+        case .onTap:
             gameScene?.deactivateGestureDetection()
         default:
```

Comment:
no defaults bro

Reply:
Tbh enum switch with default is okay mah. Since we can always guarantee that the cases are desired.

Reply:
it's the problem when new cases are added -> they just fail silently

Reply:
True